jax/api.py

```diff
@@ -1720,6 +1720,12 @@ def _jvp(fun: lu.WrappedFun, primals, tangents):
                       f"Got primal dtype {_dtype(p)} and so expected tangent dtype "
                       f"{core.primal_dtype_to_tangent_dtype(_dtype(p))}, but got "
                       f"tangent dtype {_dtype(t)} instead.")
+    try:
+      if p.shape != t.shape:
+        raise ValueError("jvp called with inconsistent primal and tangent shapes;"
+                         f"Got primal shape {p.shape} and tangent shape as {t.shape}")
+    except AttributeError:
```

Comment:
Ignoring exceptions like this is suspicious and dangerous. I assume you are trying to work around the case when `p` or `t` are Python scalars? The most general way to get a shape is with `np.shape`, which works for scalars too.

Reply:
Hi,
I was exactly trying to work around that. Thanks! for the recommendation.